.github/workflows/main.yaml

```diff
@@ -0,0 +1,48 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  HEAPS_VER: master
+
+jobs:
+  deploy-website:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # we are using this version in order to build vscode-textmate
+      - name: Setup Node
+        uses: actions/setup-node@v1
```

Comment:
Using this version is important. We can in the future move to the latest one, but we will have to figure out the whole situation with how outdated vscode-textmate is. For now let's keep it like this.